rmgpy/molecule/filtration.py

```diff
@@ -58,6 +58,9 @@ def filter_structures(mol_list, mark_unreactive=True, allow_expanded_octet=True,
     lone pairs. This function filters them out by minimizing the number of C/N/O/S atoms without a full octet.
     If ``save_order`` is ``True`` the atom order is reset after performing atom isomorphism.
     """
+    if isinstance(mol_list[0], Fragment):
+        for mol in mol_list:
+            mol.update()
```

Comment:
Question for my own understanding (no changes needed): why do fragment objects need to be updated, but not molecule objects?

Reply:
the multiplicity of Fragment is -187 by default, should be updated to avoid `ValueError` when checking multiplicity in mol_list